app/models/bank_transaction.rb

```diff
@@ -33,4 +33,56 @@ class BankTransaction < ActiveRecord::Base
   def image_url
     'data:image/png;base64,' + Base64.encode64(self.image)
   end
+
+  def assign_to_invoice
+    return false unless supplier
+
+    content = text
+    content += "\n" + reference if reference.present?
+    invoices = supplier.invoices.unpaid.select {|i| content.include? i.number}
+    invoices_sum = invoices.inject(0) {|sum,i| sum + i.amount }
+    return false if amount != -invoices_sum
+
+    ActiveRecord::Base.transaction do
+      link = FinancialLink.new
+      invoices.each {|i| i.update_attributes financial_link: link, paid_on: date}
+      update_attribute :financial_link, link
+    end
+
+    return true
+  end
+
+  def assign_to_ordergroup
+    m = /FS(?<group>\d+)(\.(?<user>\d+))?(?<parts>([A-Za-z]+\d+(\.\d+)?)+)/.match(reference)
```

Comment:
Would it be good to include some kind of checksum here? A small typo may result in a big impact, for a user. Of course the reference would then need to be generated from Foodsoft.

Reply:
NO. the format is designed that people can generate it by hand. Since the sum of the parts must match the amount of the transaction we have some kind of "checksum".